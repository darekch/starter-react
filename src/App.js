import React, { Component } from 'react';
import { hot } from 'react-hot-loader/root';

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Hello World</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
      </div>
    );
  }
}

export default hot(App);
