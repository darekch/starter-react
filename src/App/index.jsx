import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';

import Routes from './routes';
import history from './store/history';
import configureStore from './store/configureStore';

const store = configureStore();

const App = () => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Routes />
    </ConnectedRouter>
  </Provider>
);

export default App;
