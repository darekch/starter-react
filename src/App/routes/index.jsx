import React from 'react';
import { Route, Switch } from 'react-router';
import { Link } from 'react-router-dom';

import Items from 'features/Items';

const routes = () => (
  <div>
    <nav>
      <Link to="/">Home</Link>
      <Link to="/items">Items</Link>
    </nav>
    <Switch>
      <Route exact path="/items" component={Items} />
    </Switch>
  </div>
);

export default routes;
