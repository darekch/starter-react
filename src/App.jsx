import React, { Component } from 'react';
import { hot } from 'react-hot-loader/root';

import Button from 'components/Button';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Hello World</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
        <Button />
      </div>
    );
  }
}

export default hot(App);
