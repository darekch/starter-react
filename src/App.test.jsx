import React from 'react';
import ReactDOM from 'react-dom';

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import App from './App';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('snapshot', () => {
    const tree = mount(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
