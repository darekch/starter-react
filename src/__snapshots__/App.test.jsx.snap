// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<App>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "injectedReducers": Object {},
        "injectedSagas": Object {},
        "replaceReducer": [Function],
        "runSaga": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ConnectFunction
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ConnectedRouterWithContext
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        onLocationChanged={[Function]}
      >
        <ConnectedRouter
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          onLocationChanged={[Function]}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "injectedReducers": Object {},
              "injectedSagas": Object {},
              "replaceReducer": [Function],
              "runSaga": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
        >
          <Router
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
          >
            <routes>
              <div>
                <nav>
                  <Link
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      Home
                    </a>
                  </Link>
                  <Link
                    to="/asd"
                  >
                    <a
                      href="/asd"
                      onClick={[Function]}
                    >
                      Todo
                    </a>
                  </Link>
                </nav>
                <Switch />
              </div>
            </routes>
          </Router>
        </ConnectedRouter>
      </ConnectedRouterWithContext>
    </ConnectFunction>
  </Provider>
</App>
`;
